lib/ui/home/home_screen.dart

```diff
@@ -43,16 +80,75 @@ class _HomeScreenState extends ConsumerState<HomeScreen> {
           orElse: () => {});
     });
     return Scaffold(
-      body: SafeArea(
-        child: Consumer(
-          builder: (context, widgetRef, _) {
-            return HomeHeader(
-              date: DateTime.now().getFormattedString(),
-              avatar: widgetRef.watch(_profileAvatarProvider),
-            );
-          },
-        ),
+      body: Consumer(
+        builder: (context, widgetRef, _) {
+          bool isLoading = widgetRef.watch(_loadingStateProvider);
+          return isLoading
+              ? const SurveyShimmerLoading()
+              : _buildHomeContent(mockSurveys);
+        },
       ),
     );
   }
+
+  Widget _buildHomeContent(List<SurveyModel> surveys) => Consumer(
+        builder: (context, widgetRef, _) {
+          return Stack(
+            children: [
+              SurveyView(
+                surveys: surveys,
+                onPageChanged: (visibleIndex) {
+                  Future.delayed(const Duration(milliseconds: 100), () {
+                    ref.read(_visibleIndexProvider.notifier).state =
+                        visibleIndex;
+                  });
+                },
+                onSurveySelected: (survey) {
+                  // TODO: Navigate to survey details
+                },
+              ),
+              SafeArea(
+                child: HomeHeader(
+                  date: DateTime.now().getFormattedString(),
+                  avatar: widgetRef.watch(_profileAvatarProvider),
+                ),
+              ),
+              _pagerIndicator(mockSurveys)
+            ],
+          );
+        },
+      );
+
+  Widget _pagerIndicator(List<SurveyModel> surveys) => Consumer(
```

Comment:
@hoangnguyen92dn I think we can create this in a separate class, maybe `PageIndicatorWidget` 🤔

Reply:
@doannimble It was my thought, but we don't reuse this widget outside home screen so it still makes sense to leave it here 😄

Reply:
@hoangnguyen92dn In Flutter, we recommend creating a new Widget, not only for reuse purposes: https://www.youtube.com/watch?v=IOyq-eTRhvo. Besides, it would help us to reduce the complexity of the main widget, the LOC 🤔

Reply:
🙆‍♂️